### ISSUE TITLE
How do additional metrics function in train_model?

### ISSUE THREAD

**Issue by NienkeMekkes** [ThilinaRajapakse/simpletransformers#1358]
Dear @ThilinaRajapakse ,

Thank you for the great tool. I hope you have time for a question regarding additional metrics.

I have a multilabel classification task.
Since the task is based on imbalanced data, I am curious about how the performance would be if I use the macro F1 score as an additional metric to train on. However, I get confused when trying out if this is possible

Inspired by issue  [125](https://github.com/ThilinaRajapakse/simpletransformers/issues/125) I have written:

from sklearn.preprocessing import MultiLabelBinarizer

def f1(d,predictions):
     mlb = MultiLabelBinarizer()
    
    return f1_score(
        mlb.fit_transform(d['labels']),
        mlb.fit_transform(predictions),
        average='macro'
    )

I define a model using MultiLabelClassificationModel() and then train using:
model.train_model(train,eval_df = val,f1=f1)

The code executes without any errors.
I then make predictions on the same evaluation file using model.predict(), and check how well my model performed. However, if I run training with the additional f1 metric or without it, the predictions are always identical. How do I see and understand the influence of the additional metric during training?

Thank you,
Nienke
